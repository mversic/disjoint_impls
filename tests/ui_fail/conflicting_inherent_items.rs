use disjoint_impls::disjoint_impls;

pub trait Dispatch {
    type Group;
}

pub enum GroupA {}
impl Dispatch for String {
    type Group = GroupA;
}

pub enum GroupB {}
impl Dispatch for i32 {
    type Group = GroupB;
}

struct Wrapper<T>(T);

disjoint_impls! {
    impl<T: Dispatch<Group = GroupA>> Wrapper<T> {
        fn kita() -> &'static str {
            "Blanket A"
        }
    }
    impl<T: Dispatch<Group = GroupB>> Wrapper<T> {
        fn kita() -> String {
            "Blanket B".to_owned()
        }
    }
}

fn main() {}
