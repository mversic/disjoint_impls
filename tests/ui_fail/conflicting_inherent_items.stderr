error[E0592]: duplicate definitions with name `kita`
  --> tests/ui_fail/conflicting_inherent_items.rs:21:9
   |
21 |         fn kita() -> &'static str {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^ duplicate definitions for `kita`
...
26 |         fn kita() -> String {
   |         ------------------- other definition for `kita`
